src/main/java/org/commcare/formplayer/session/FormSession.java

```diff
@@ -440,64 +394,27 @@ private void deserializeFormDef(String serializedFormDef) throws IOException, De
 
 
     public SerializableFormSession serialize() throws IOException {
-        SerializableFormSession serializableFormSession = new SerializableFormSession();
-        serializableFormSession.setInstanceXml(getInstanceXml());
-        serializableFormSession.setId(getSessionId());
-        serializableFormSession.setFormXml(serializeFormDef());
-        serializableFormSession.setUsername(username);
-        serializableFormSession.setSessionData(getSessionData());
-        serializableFormSession.setSequenceId(getSequenceId());
-        serializableFormSession.setInitLang(getLocale());
-        serializableFormSession.setSessionData(getSessionData());
-        serializableFormSession.setDomain(getDomain());
-        serializableFormSession.setPostUrl(getPostUrl());
-        serializableFormSession.setMenuSessionId(menuSessionId);
-        serializableFormSession.setTitle(getTitle());
-        serializableFormSession.setDateOpened(new Date().toString());
-        serializableFormSession.setOneQuestionPerScreen(oneQuestionPerScreen);
-        serializableFormSession.setCurrentIndex(currentIndex);
-        serializableFormSession.setAsUser(asUser);
-        serializableFormSession.setAppId(appId);
-        serializableFormSession.setFunctionContext(functionContext);
-        serializableFormSession.setInPromptMode(inPromptMode);
-        serializableFormSession.setRestoreAsCaseId(restoreAsCaseId);
-        return serializableFormSession;
-    }
-
-    public String getDomain() {
-        return domain;
+        session.incrementSequence();
```

Comment:
This is the only potentially meaningful semantic change I found in the PR. It looks like this changes the overall approach slightly from relying on the user to increment at the right time (specifically when `updateSession` was called), to presuming that any time the session data is serialized we'll increment the counter so that what is written has a new serial count? That sounds right to me, but wanted to make sure I was seeing it correctly.

Reply:
That's correct, currently the sequence ID is only used at the end of the request to send in the response which is after the session is saved. I agree that if this function were called more than once per request we'd double increment.

I could revert this to setting the value to `0` initially and then only incrementing in `updateSession`.

Reply:
Cool, and I don't think it's an issue if the number gets incremented more often than necessary, just making sure I see the path through